cadquery/occ_impl/exporters/assembly.py

```diff
@@ -99,6 +105,169 @@ def exportAssembly(
     return status == IFSelect_ReturnStatus.IFSelect_RetDone
 
 
+def exportMetaStep(
+    assy: AssemblyProtocol,
+    path: str,
+    write_pcurves: bool = True,
+    precision_mode: int = 0,
+) -> bool:
+    """
+    Export an assembly to a STEP file with faces tagged with names and colors. This is done as a
+    separate method from the main STEP export because this is not compatible with the fused mode
+    and also flattens the hierarchy of the STEP.
+
+    Layers are used because some software does not understand the ADVANCED_FACE entity and needs
+    names attached to layers instead.
+
+    :param assy: assembly
+    :param path: Path and filename for writing
+    :param names: A dictionary of that specifies faces that need to have string names attached.
+    :param colors: A dictionary of that specifies faces that need to have RGB colors attached.
+    :param layers: A dictionary of that specifies faces that need to have string layer names attached.
+    :param write_pcurves: Enable or disable writing parametric curves to the STEP file. Default True.
+        If False, writes STEP file without pcurves. This decreases the size of the resulting STEP file.
+    :param precision_mode: Controls the uncertainty value for STEP entities. Specify -1, 0, or 1. Default 0.
+        See OCCT documentation.
+    """
+
+    pcurves = 1
+    if not write_pcurves:
+        pcurves = 0
+
+    # Initialize the XCAF document that will allow the STEP export
+    app = XCAFApp_Application.GetApplication_s()
+    doc = TDocStd_Document(TCollection_ExtendedString("XmlOcaf"))
+    app.InitDocument(doc)
+
+    # Shape and color tools
+    shape_tool = XCAFDoc_DocumentTool.ShapeTool_s(doc.Main())
+    color_tool = XCAFDoc_DocumentTool.ColorTool_s(doc.Main())
+    layer_tool = XCAFDoc_DocumentTool.LayerTool_s(doc.Main())
+
+    def _process_child(child: AssemblyProtocol, assy_label: TDF_Label):
+        """
+        Process a child part which is not a subassembly.
+        :param child: Child part to process (we should already have filtered out subassemblies)
+        :param assy_label: The label for the assembly to add this part to
+        :return: None
+        """
+
+        # We combine these because the metadata could be stored at the parent or child level
+        combined_names = {**assy.subshape_names, **child.subshape_names}
+        combined_colors = {**assy.subshape_colors, **child.subshape_colors}
+        combined_layers = {**assy.subshape_layers, **child.subshape_layers}
+
```

Comment:
`child_items = None`

Reply:
`child_items` is now set to `None` in b467f342e6cdf32855054c77c25c45cc4171bdf6 . I'll wait to see what the test coverage looks like with the latest commit before resolving this thread.

Reply:
I already have a test case that was designed to cover this line [here](https://app.codecov.io/gh/CadQuery/cadquery/pull/1782?dropdown=coverage&src=pr&el=h1&utm_medium=referral&utm_source=github&utm_content=checks&utm_campaign=pr+comments&utm_term=CadQuery#7d124cdb7ef86c9612cc78b4e996495d-R163), but Codecov still says it is only partially covered. Test is [here](https://github.com/CadQuery/cadquery/pull/1782/files#diff-6aee0cfc95b61e4b6a586449172cfbe844a992d6e1826cf459d7690c583ad939R745).

Reply:
Ok, it looks like I have that partial coverage fixed now.